src/js/utils/MesosStateUtil.js

```diff
@@ -2,8 +2,34 @@ import Util from './Util';
 
 const RESOURCE_KEYS = ['cpus', 'disk', 'mem'];
 
+function setSchedulerName(name, tasks) {
+  return tasks.map(function (task) {
+    let newTask = Object.assign({}, task);
+
+    newTask.scheduler = name;
+
+    return newTask;
+  });
+}
+
 const MesosStateUtil = {
 
+  assignSchedulerNameToTasks(state) {
```

Comment:
Where is that feature needed? In the task table? And how are you then going to get the icon from that framework?

Anyways, the only reliable way of telling which framework it belongs to, is through the `framework_id` property and find the framework like this: https://github.com/dcos/dcos-ui/blob/master/src/js/pages/task-details/TaskDetailsTab.js#L41

If that is going to go in the task table I would strongly encourage that we revisit this solution and instead make the icon that we need available on the task when we get the (root) `marathon` groups information, where the icon information is coming from.